### PR TITLE
add task6

### PR DESCRIPTION
**Pull request by AndreyNikitsich**

### CODE REVIEW

File: UI/script.js [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+            discount: "5%",
+            validUntill: new Date("2021-05-01")
+        },
+        ];
+        
+    function cmpByDate(a,b) {
+        return a.createdAt - b.createdAt;
+    }  
+
+    Array.prototype.contains = function(array) {
```
Comment (YauheniyaBasaranovich) [inline UI/script.js:210]: Do you really need to overwrite this method? Couldn't required functionality be achieved with just a regular function?

File: UI/script.js [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+    }
+
+    function removeAd(id){
+        var toDelete = adList.findIndex(function(item,index,arr){
+            return item.id===id;
+        })
+
+        if(toDelete===-1){
+            return false;
+        }else{
```
Comment (YauheniyaBasaranovich) [inline UI/script.js:259]: No need to use `else` after return

File: UI/script.js [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+                }
+            }
+            return true;
+        }
+    }
+
+    function validateAd(ad){
+        var valid = false;
+        if(!ad.id || typeof ad.id !== "string"){
+
```
Comment (YauheniyaBasaranovich) [inline UI/script.js:285]: You can return `false` everytime you meet invalid fields and `true` in the end of function

**Review by YauheniyaBasaranovich — APPROVED** [review]
Approved, although I see a lot of files which should be part of task 3.

**Comment by AndreyNikitsich** [comment]
Thanks for reply. I've fixed these things in task7

**Pull request merged**